crates/noirc_frontend/src/hir_def/types.rs

```diff
@@ -589,9 +604,33 @@ impl Type {
         }
     }
 
+    fn get_fixed_variable_array_length(&self, ident: &Ident, interner: &NodeInterner) -> u128 {
+        let stmt_id = interner
+            .get_global_const(ident)
+            .expect("no statement associated with fixed sized array variable");
+        let statement = interner.statement(&stmt_id);
+        match statement {
+            HirStatement::Let(let_stmt) => {
+                let expression = interner.expression(&let_stmt.expression);
+                match expression {
+                    HirExpression::Literal(literal) => {
```

Comment:
Perhaps the global_consts should store the expression directly instead of the statement id and matching on the statement each time to extract the let_stmt.expression. If we also require them to be integers we could go one step further and store only the integer. Then the FixedVariable array length wouldn't be needed

Reply:
I had made the change to storing an expr id rather than stmt. I reverted back to the stmt id for now as I needed it when calling `add_global_variable_decl` (which can be found here: https://github.com/noir-lang/noir/pull/319/files#diff-a9bedea75b438a9e39c699dfd6a9305a19cbd11cce1d562405f30633c1be5865R174) in `resolve_pattern`. In order not to have duplicate definition IDs I check whether a global const stmt has already been interned and fetch the existing `HirIdent` to use in the global scope for an internal function resolver. This is also why I decided to intern the ident for global consts, as it was easier to fetch the global const directly. However, I want to rethink this as you mentioned in this comment (https://github.com/noir-lang/noir/pull/319/files#r959636365), we want differently named consts in different modules. 

While I am thinking about how to reorganize `collect_global_consts` and `resolve_global_consts` so that I am not interning multiple statements, I will also consider if there is a way that I can avoid interning the stmt id and I can use expr id instead.  

I was thinking perhaps we shouldn't go to the final step though and store only the integer. It seems that will have to potentially be changed back to an expr or stmt being stored if we ever want to store const arrays or bools. Perhaps we could store only the literal. But then come type checking for the fixed variable array length would still need to be done.

Reply:
I made a change to have a `GlobalConstInfo` type that stores the `Ident` (to easily fetch the name of a const), `ExprId` (easily fetch the rhs expr of a statement), and `LocalModuleId` (this scopes global consts to one module).